src/WaveClip.cpp

```diff
@@ -1212,51 +1212,43 @@ void WaveClip::Flush()
    //wxLogDebug(wxT("now sample count %lli"), (long long) mSequence->GetNumSamples());
 }
 
-bool WaveClip::HandleXMLTag(const wxChar *tag, const wxChar **attrs)
+bool WaveClip::HandleXMLTag(const std::string_view& tag, const AttributesList &attrs)
 {
-   if (!wxStrcmp(tag, wxT("waveclip")))
+   if (tag == "waveclip")
    {
       double dblValue;
       long longValue;
-      while (*attrs)
+      for (auto pair : attrs)
       {
-         const wxChar *attr = *attrs++;
-         const wxChar *value = *attrs++;
+         auto attr = pair.first;
+         auto value = pair.second;
 
-         if (!value)
-            break;
-
-         const wxString strValue = value;
-         if (!wxStrcmp(attr, wxT("offset")))
+         if (attr == "offset")
          {
-            if (!XMLValueChecker::IsGoodString(strValue) ||
-                  !Internat::CompatibleToDouble(strValue, &dblValue))
+            if (!value.TryGet(dblValue))
                return false;
             SetSequenceStartTime(dblValue);
          }
-         else if (!wxStrcmp(attr, wxT("trimLeft")))
+         else if (attr == "trimLeft")
          {
-            if (!XMLValueChecker::IsGoodString(strValue) ||
-               !Internat::CompatibleToDouble(strValue, &dblValue))
+            if (!value.TryGet(dblValue))
                return false;
             SetTrimLeft(dblValue);
          }
-         else if (!wxStrcmp(attr, wxT("trimRight")))
+         else if (attr == "trimRight")
          {
-            if (!XMLValueChecker::IsGoodString(strValue) ||
-               !Internat::CompatibleToDouble(strValue, &dblValue))
+            if (!value.TryGet(dblValue))
                return false;
             SetTrimRight(dblValue);
          }
-         else if (!wxStrcmp(attr, wxT("name")))
+         else if (attr == "name")
          {
-            if(XMLValueChecker::IsGoodLongString(strValue))
```

Comment:
I just checked again the definitions of IsGoodString and IsGoodLongString.  The first had an additional check that the other did not, that a string isn't over 4096 characters and possibly from a malicious file, or so the comment said.

Are you intentionally removing the excessive length check?  I don't see where in the PR it is preserved.

See commit 7547de9 which was recent and relevant, changing the bound used for the check.  Although in that commit, pehaps the better thing would have been to use IsGoodLongString somewhere to avoid the check.

Reply:
IsGoodLongString() only checks that string has no `\0` except for the null terminator. It probably was relevant someday, but I don't see any point in keeping it. 

IsGoodString() exposes some arbitrary global constraint on all the strings originating from the XML file. I'm glad it is no longer used because such constraints should be implemented in a way that informs the user about it, not simply by discarding the data during the parsing. 

Before https://github.com/audacity/audacity/commit/7547de9f5298a4fb710c1a44069e10dbc982c2e0 the situation was especially bad because projects would open differently on different platforms.

My point is - I see no reasoning behind checking for null characters or why the limit of 4096 characters is used.

The first check is redundant by construction. Expat is a C library, and it could not produce a string with null characters in the middle.